File: php/PersonagemConcreteBuilder.php
<?php

class PersonagemConcreteBuilder implements PersonagemBuilder{

    private $personagem;

    public function __construct()
    {
        $this->personagem = new Personagem();
    }

    public function buildClasse()
    {
        $this->personagem->setClasse("Caçador");
    }

    public function buildEspecie()
    {
        $this->personagem->setEspecie("Desperto");
    }

    public function buildNome()
    {
        $this->personagem->setNome("DeathlKaileena");
    }

    public function getPersonagem()
    {
        return $this->personagem;
    }
}

File: java/PersonagemConcreteBuilder.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package personagem;

/**
 *
 * @author User
 */
public class PersonagemConcreteBuilder implements PersonagemBuilder {
    
    private Personagem personagem;

    public PersonagemConcreteBuilder() {
        this.personagem = new Personagem();
    }
    
    
    @Override
    public void buildClasse() {
        personagem.setClasse("Caçador");
    }

    @Override
    public void buildEspecie() {
        personagem.setEspecie("Exo");
    }

    @Override
    public void buildNome() {
        personagem.setNome("Cayde-6");
    }

    @Override
    public Personagem getPersonagem() {
        return personagem;
    }
    
}
